### PR TITLE
Login logout w redirect dash/landing

### PR DESCRIPTION
**Pull request by sdebaun**

### CODE REVIEW

File: js/components/Landing.js [hunk 1]
```diff
@@ -106,5 +108,7 @@ export default (sources) => {
 
   return {
     DOM: Observable.just(view),
+    auth$: appMenu.auth$,
+    route$: auth$.filter(auth => !!auth).map(() => '/dash'),
```
Comment (TylorS) [inline js/components/Landing.js:112]: :+1:

**Comment by sdebaun** [comment]
hehehe i was wondering if any of the extremely terse code i was writing was annoying you :)

for some reason im finding that breaking things down into model/view/intent functions is not coming to me naturally...

**Comment by TylorS** [comment]
When I've played around with writing applications, and what I've recommended to others, is to think about what your model will contain _first_.

What state does this component need to have or understand in order to create it's view? Once that is decided I like to just mock the state$ of the component.

``` js

// simple mock of state$
const state$ = Observable.of({
  auth: {user: 'sdebaun', isAuthenticated: true},
  isOpen: true
})
```

Now with the mock state$ you're able to create your view with all the information that it can be provided.

``` js

// can be wrapped into a separate view(state$) function 
const view$ = state$.map(state => {
  return div([
    ...
  ])
})

return {
  DOM: view$
  ...
}
```

From there you have a working view that displays the state, but it doesn't _do_ anything.

So then you work through your intent function to actually listen to stuff that happens.

``` js
// IMO intent *only* comes from user interaction
// which in a browser is really only interaction with the DOM
function intent(DOM) {
  const click$ = DOM.select('.button').events('click')

  const actions = {
    click$,
  }

  return actions // just to illustrate my naming choices, could return object literal
}
```

With your intent function now written you're able to create a proper model function to produce your state$.

``` js
// actions are what are returned from intent()
function model(actions, sources) {

  const auth$ = sources.HTTP.filter(res.req.url === '/auth')
    .map(req => req.body)
  const isOpen$ = actions.click$.map(({target}) => target.something)

  const state$ = Observable.combineLatest(
    auth$, isOpen$,
    (auth, isOpen) => ({auth, isOpen})
  )

  return state$
}
```

Then you're free to replace state$ with your model function and you have yourself a fully built component.

**Comment by TylorS** [comment]
So you end up with something like

``` js
function Component(sources) {
  const actions = intent(sources.DOM)
  const state$ = model(actions, sources)
  const view$ = view(state$)

  return {
    DOM: view$,
    auth$: state$.pluck('auth'),
    isOpen$: state$.pluck('isOpen'),
  }
}
```

**Pull request merged**